app/src/main/java/org/oppia/app/story/StoryViewModel.kt

```diff
@@ -0,0 +1,76 @@
+package org.oppia.app.story
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.ChapterSummary
+import org.oppia.app.model.StorySummary
+import org.oppia.domain.topic.TopicController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+
+@FragmentScope
+class StoryViewModel @Inject constructor(
+  private val topicController: TopicController,
+  private val logger: Logger
+) : ViewModel() {
+  private lateinit var storyId: String
+
+  val storyLiveData: LiveData<StorySummary> by lazy {
+    processStoryLiveData()
+  }
+
+  val storyNameLiveData: LiveData<String> by lazy {
+    processStoryNameLiveData()
+  }
+
+  val storyChapterLiveData: LiveData<List<ChapterSummary>> by lazy {
+    processStoryChapterLiveData()
+  }
+
+  private val storyResultLiveData: LiveData<AsyncResult<StorySummary>> by lazy {
+    topicController.getStory(storyId)
+  }
+
+  fun setStoryId(storyId: String) {
+    this.storyId = storyId
+  }
+
+  private fun processStoryLiveData(): LiveData<StorySummary> {
+    return Transformations.map(storyResultLiveData, ::processStoryResult)
```

Comment:
You could also process the storyResultLiveData so that it's a ``LiveData<StorySummary>`` rather than the AsyncResult of the story summary. Then you wouldn't need to check for the error case in each of the transformed methods.

You can also probably get rid of the extra process methods by leveraging method references, e.g.:

```kotlin
val storyNameLiveData: LiveData<String> by lazy {
  Transformations.map(storyResultLiveData, StorySummary::getStoryName)
}
```

Reply:
Removed the extra process methods.
Also, I didn't understand the initial part about removing error checks. Can you tell that again?

Reply:
Done.